reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -942,7 +941,66 @@
 	 * @return a new {@link Flux} behaving like the fastest of its sources
 	 */
 	public static <I> Flux<I> first(Iterable<? extends Publisher<? extends I>> sources) {
-		return onAssembly(new FluxFirstEmitting<>(sources));
+		return onAssembly(new FluxFirstSignalling<>(sources));
+	}
+
+	/**
+	 * Pick the first {@link Publisher} to emit any value and replay all values
+	 * from that {@link Publisher}, effectively behaving like the source that first
+	 * emits an {@link Subscriber#onNext(Object) onNext}.
+	 *
+	 * <p>
+	 * // TODO replace the img
+	 * <p>
+	 * Valued sources always "win" over an empty source (one that only emits onComplete)
+	 * or a failing source (one that only emits onError).
+	 * Note that like in {@link #first(Iterable)}, an infinite source can be problematic
+	 * if no other source emits onNext.
+	 *
+	 * @param sources An {@link Iterable} of the competing source publishers
+	 * @param <I> The type of values in both source and output sequences
+	 *
+	 * @return a new {@link Flux} behaving like the fastest of its sources
+	 */
+	public static <I> Flux<I> firstValued(Iterable<? extends Publisher<? extends I>> sources) {
```

Comment:
this sounds really weird to me ("valued" as an adjective, not verb)

What about `race`/`fastest`/other suggested names from the issue?